### PR TITLE
🐛✨EC2 machine creation: try to create at least 1 machine

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

**BEFORE:** The autoscaling would try to create up to ```EC2_INSTANCES_MAX_INSTANCES``` instances and fail to do so if for example the EC2 number of machines quota is exceeded.
**AFTER:** The autoscaling will try to create between 1 and ```EC2_INSTANCES_MAX_INSTANCES``` instances. so if the quota is exceeded, EC2 will try to create between 1 and the asked number of instances.

## Related issue/s
- fixes https://github.com/ITISFoundation/osparc-issues/issues/1277

## Dev Checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]
nice :100:

**Comment by mguidon** [comment]
Is there any useful log message reported back (through the API server) if we run out of quota?

**Comment by sanderegg** [comment]
> Is there any useful log message reported back (through the API server) if we run out of quota?

@mguidon : 
```
Unexpected error happened when starting EC2 instance: An error occurred (VcpuLimitExceeded) when calling the RunInstances operation: You have requested more vCPU capacity than your current vCPU limit of 1000 allows for the instance bucket that the specified instance type belongs to. Please visit http://aws.amazon.com/contact-us/ec2-request to request an adjustment to this limit.
```

**Review by mguidon — APPROVED** [review]

**Review by GitHK — APPROVED** [review]

File: packages/aws-library/src/aws_library/ec2/client.py [hunk 1]
```diff
@@ -131,7 +131,9 @@ async def start_aws_instance(
 
             instances = await self.client.run_instances(
                 ImageId=instance_config.ami_id,
-                MinCount=number_of_instances,
+                MinCount=min(
+                    1, number_of_instances
+                ),  # NOTE: this ensures as many instances as possible are created
```
Comment (pcrespov) [inline packages/aws-library/src/aws_library/ec2/client.py:136]: I am not sure I understand this note. 

As far as I understand now the "Count" is set between 
`min(1, number_of_instances)< x < number_of_instances`  but not sure what the implications are.

I wonder what is the different between your option and `1 < x < number_of_instances` provided that possible number_of_instances is probably at least one?

**Review by pcrespov — APPROVED** [review]

File: packages/aws-library/src/aws_library/ec2/client.py (hunk 1 above)
Comment (sanderegg) [reply packages/aws-library/src/aws_library/ec2/client.py:136]: adapted

**Pull request merged**